modules/beta-autopilot-private-cluster/cluster.tf

```diff
@@ -0,0 +1,219 @@
+    content {
+      type = var.cluster_telemetry_type
+    }
+  }
+  logging_service    = local.cluster_telemetry_type_is_set ? null : var.logging_service
+  monitoring_service = local.cluster_telemetry_type_is_set ? null : var.monitoring_service
+  vertical_pod_autoscaling {
+    enabled = var.enable_vertical_pod_autoscaling
+  }
+  enable_autopilot = var.enable_autopilot
+  dynamic "master_authorized_networks_config" {
+    for_each = local.master_authorized_networks_config
+    content {
+      dynamic "cidr_blocks" {
+        for_each = master_authorized_networks_config.value.cidr_blocks
+        content {
+          cidr_block   = lookup(cidr_blocks.value, "cidr_block", "")
+          display_name = lookup(cidr_blocks.value, "display_name", "")
+        }
+      }
+    }
+  }
+
+  master_auth {
+    username = var.basic_auth_username
+    password = var.basic_auth_password
+
+    client_certificate_config {
+      issue_client_certificate = var.issue_client_certificate
+    }
+  }
+
+  addons_config {
+    http_load_balancing {
+      disabled = ! var.http_load_balancing
+    }
+
+    horizontal_pod_autoscaling {
+      disabled = ! var.horizontal_pod_autoscaling
+    }
+
+    istio_config {
+      disabled = ! var.istio
+      auth     = var.istio_auth
+    }
+
+    dynamic "cloudrun_config" {
+      for_each = local.cluster_cloudrun_config
+
+      content {
+        disabled = cloudrun_config.value.disabled
+      }
+    }
+
+    dns_cache_config {
+      enabled = var.dns_cache
+    }
+
+    dynamic "gce_persistent_disk_csi_driver_config" {
+      for_each = local.cluster_gce_pd_csi_config
+
+      content {
+        enabled = gce_persistent_disk_csi_driver_config.value.enabled
+      }
+    }
+
+    kalm_config {
+      enabled = var.kalm_config
+    }
+
+    config_connector_config {
+      enabled = var.config_connector
+    }
+  }
+  datapath_provider = var.datapath_provider
+
+  networking_mode = "VPC_NATIVE"
+  ip_allocation_policy {
+    cluster_secondary_range_name  = var.ip_range_pods
+    services_secondary_range_name = var.ip_range_services
+  }
+
+  maintenance_policy {
+    dynamic "recurring_window" {
+      for_each = local.cluster_maintenance_window_is_recurring
+      content {
+        start_time = var.maintenance_start_time
+        end_time   = var.maintenance_end_time
+        recurrence = var.maintenance_recurrence
+      }
+    }
+
+    dynamic "daily_maintenance_window" {
+      for_each = local.cluster_maintenance_window_is_daily
+      content {
+        start_time = var.maintenance_start_time
+      }
+    }
+
+    dynamic "maintenance_exclusion" {
+      for_each = var.maintenance_exclusions
+      content {
+        exclusion_name = maintenance_exclusion.value.name
+        start_time     = maintenance_exclusion.value.start_time
+        end_time       = maintenance_exclusion.value.end_time
+      }
+    }
+
+  }
+
+  lifecycle {
+    ignore_changes = [node_pool, initial_node_count]
```

Comment:
do we need to ignore changes? I believe this was workaround for permadiff with initial nodepools in standard mode